plugin/advisor/mysql/advisor_stmt_no_leading_wildcard_like.go

```diff
@@ -0,0 +1,99 @@
+package mysql
+
+import (
+	"fmt"
+
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/format"
+)
+
+const (
+	wildcard byte = '%'
+)
+
+var (
+	_ advisor.Advisor = (*NoLeadingWildcardLikeAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLNoLeadingWildcardLike, &NoLeadingWildcardLikeAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLNoLeadingWildcardLike, &NoLeadingWildcardLikeAdvisor{})
+}
+
+// NoLeadingWildcardLikeAdvisor is the advisor checking for no leading wildcard LIKE.
+type NoLeadingWildcardLikeAdvisor struct {
+}
+
+// Check checks for no leading wildcard LIKE.
+func (adv *NoLeadingWildcardLikeAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &noLeadingWildcardLikeChecker{level: level}
+	for _, stmtNode := range root {
+		checker.text = stmtNode.Text()
+		checker.leadingWildcardLike = false
+		(stmtNode).Accept(checker)
+
+		if checker.leadingWildcardLike {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.StatementLeadingWildcardLike,
+				Title:   "No leading wildcard LIKE",
+				Content: fmt.Sprintf("%q uses leading wildcard LIKE", checker.text),
+			})
+		}
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+
+}
+
+type noLeadingWildcardLikeChecker struct {
+	adviceList          []advisor.Advice
+	level               advisor.Status
+	text                string
```

Comment:
Seems no need to have the `text` property? We can use `stmtNode.Text()` in line 53, and `in.Text()` in line 86

Reply:
A sad message is we can only use `in.Text()` when `in` is the root node in `AST`. Otherwise, it will return an empty string.
The `text` property does be used for line 86.